test/HelmetDeclarativeTest.js

```diff
@@ -2501,6 +2501,41 @@ describe("Helmet - Declarative API", () => {
         });
     });
 
+    describe("deferred tags", () => {
+        beforeEach(() => {
+            window.__spy__ = sinon.spy();
+        });
+
+        afterEach(() => {
+            delete window.__spy__;
+        });
+
+        it("executes synchronously when defer={true} and async otherwise", done => {
+            ReactDOM.render(
+                <Helmet>
+                    <script defer={false}>
```

Comment:
Option 2 would be to keep this syntax, but strip out invalid `defer` attributes before we write to the DOM.